mod_reforged/hooks/items/weapons/weapon.nut

```diff
@@ -57,8 +68,99 @@
 		}
 	}
 
+// New Functions
 	q.getReach <- function()
 	{
 		return this.m.Reach;
 	}
+
+	q.assignDefaultReach <- function()
```

Comment:
The logic in this function needs to be improved in a way that hybrid weapons are handled better. E.g. currently it will set a Dagger/Sword to 1 reach because it checks for Dagger first even though it might be a weapon similar to the Ancient Sword in Reforged.

Alternative logic could be:
- Start with 0 Reach.
- Each detected weapon type adds its own Reach value. E.g. Dagger adds 1, Axe adds 3, Sword adds 4, Spear adds 5 and so on.
- Then divide the total Reach by the total number of Weapon Types detected rounded to the nearest integer.
- Being Two-Handed adds 1 Reach if it isn't already at 5 Reach by now.
- Having a MaxRange of 2 adds 1 Reach if it isn't already at 6 Reach by now.

So in this case a Dagger/Sword will end up with 1+4 = 5 / 2 = 2.5 = 3 Reach. An Axe/Sword will end up with 3 + 4 = 7 / 2 = 3.5 = 4 Reach. An Axe/Hammer will end up with 3 + 3 = 6 / 2 = 3 Reach. And so on.

Reply:
I would like to pass this advanced default reach algorithm implementation onto you or someone else. Maybe do it in another branch?

Reply:
Ok. I'll implement it in a separate branch.

Reply:
My implementation is now up in #526